### PR TITLE
Added SeiManga extension

### LINKED ISSUE #1849: Source request: SeiManga
### Source name

SeiManga

### Source link

https://seimanga.me/

### Source language

Russian

### Other details

18+/NSFW = yes
Seinen, josei

### Acknowledgements

- [X] I have checked that the extension does not already exist by searching the [extension listing](https://keiyoushi.github.io/extensions/) and verified it does not appear there.
- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a meaningful title with the source name.
- [ ] I have admitted that I am a clown by having checked this box, as I have not read these acknowledgements.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by Teashrock**
Closes #1849. 

Checklist:

- [X] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [X] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [X] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [X] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [X] Have not changed source names
- [X] Have explicitly kept the `id` if a source's name or language were changed
- [X] Have tested the modifications by compiling and running the extension through Android Studio
- [X] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

File: src/ru/seimanga/build.gradle [hunk 1]
```diff
@@ -0,0 +1,10 @@
+ext {
+    extName = 'SeiManga'
+    extClass = '.SeiManga'
+    themePkg = 'grouple'
+    baseUrl = 'https://seimanga.me'
+    overrideVersionCode = 46
```
Comment (AwkwardPeak7) [inline src/ru/seimanga/build.gradle]:
```suggestion
    overrideVersionCode = 0
```

File: src/ru/seimanga/src/eu/kanade/tachiyomi/extension/ru/seimanga/SeiManga.kt [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import okhttp3.Request
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+
+class SeiManga : GroupLe("SeiManga", "https://seimanga.me", "ru") {
+
+    override val id: Long = 6
```
Comment (AwkwardPeak7) [inline src/ru/seimanga/src/eu/kanade/tachiyomi/extension/ru/seimanga/SeiManga.kt]:
```suggestion
```
id should only be overriden if old one needs to be preserved. let it generate automatically

**Review by AwkwardPeak7 — CHANGES_REQUESTED** [review]

File: src/ru/seimanga/src/eu/kanade/tachiyomi/extension/ru/seimanga/SeiManga.kt [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+            dialogTitle = DOMAIN_TITLE
+            setOnPreferenceChangeListener { _, newValue ->
+                try {
+                    val res = preferences.edit().putString(DOMAIN_TITLE, newValue as String).memit()
+                    Toast.makeText(screen.context, "Для смены домена необходимо перезапустить приложение с полной остановкой.", Toast.LENGTH_LONG).show()
+                    res
+                } catch (e: Exception) {
+                    e.printStackTrace()
+                    false
+                }
```
Comment (AwkwardPeak7) [inline src/ru/seimanga/src/eu/kanade/tachiyomi/extension/ru/seimanga/SeiManga.kt]:
```suggestion
                Toast.makeText(screen.context, "Для смены домена необходимо перезапустить приложение с полной остановкой.", Toast.LENGTH_LONG).show()
                true
```
build failed here btw, also you don't need to manually set the preference value, app will take care of it.

File: src/ru/seimanga/src/eu/kanade/tachiyomi/extension/ru/seimanga/SeiManga.kt [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+        Genre("эротика", "el_1340"),
+        Genre("этти", "el_1354"),
+        Genre("юри", "el_1315"),
+        Genre("яой", "el_1336"),
+    )
+
+    override fun setupPreferenceScreen(screen: androidx.preference.PreferenceScreen) {
+        super.setupPreferenceScreen(screen)
+        EditTextPreference(screen.context).apply {
+            key = DOMAIN_TITLE
```
Comment (AwkwardPeak7) [inline src/ru/seimanga/src/eu/kanade/tachiyomi/extension/ru/seimanga/SeiManga.kt]: while you are at it, change the domain preference implementation to [something like this](https://github.com/keiyoushi/extensions-source/blob/fea4ac2a754fc1d03b43eaaacb544854ba2a4865/src/en/hiperdex/src/eu/kanade/tachiyomi/extension/en/hiperdex/Hiperdex.kt#L40-L68) so domain update in extension code change the pref for users too

**Review by AwkwardPeak7 — CHANGES_REQUESTED** [review]
> Seriously, what compiler in the world would even consider a blank line an error?

Thats the linter for you 😂

File: src/ru/seimanga/src/eu/kanade/tachiyomi/extension/ru/seimanga/SeiManga.kt (hunk 3 above)
Comment (Teashrock) [reply src/ru/seimanga/src/eu/kanade/tachiyomi/extension/ru/seimanga/SeiManga.kt]: What's going on? The reason of the error is simple: it changed "commit" function name to "memit" for some reason. The latter function doesn't exist. What's even stranger is that everything's fine in my edited code, but not in the commit.

**Pull request merged**